src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
-    ArcLengthParametrization_t arc_length_parameterization,
-    const float arc_segment_length);
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_createForwardsContinuousSpeedProfile(
+    float speeds[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float max_allowable_acceleration, float initial_speed, float final_speed,
+    unsigned int num_elements);
 
 /**
- * Returns the maximum velocity profile for a given curve based on curvature and maximum
- * allowable acceleration
+ * Function generates the absolute maximum speed that can occur at any points along the
+ * trajectory. This maximum speed is determined by the limit of grip (in acceleration)
+ * along any curved path, along with imposed speed limits of the user.
  *
- * @pre max_allowable_speed_profile is pre-allocated up at least the limit specified by
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ * @pre All arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_SEGMENTS
  *
- * @param max_allowable_speed_profile [out] The pre-allocated array that will contain all
- * of the maximum allowable velocity values for each point on the curve
+ * @param path [in] The 2d polynomial that defines the path in 2d space.
  *
- * @param path [in] The polynomial that defines the path of the trajectory
+ * @param num_elements [in] The number of elements that will make up the trajectory
  *
- * @param num_elements [in] The number of elements in the trajectory
+ * @param t_start [in] The starting value of the parameterization
  *
- * @param arc_length_parameterization [in] The arc length parameterization of the path
- * polynomial
+ * @param t_end [in] The ending value of the parameterization
  *
- * @param arc_segment_length [in] The length of each segment in the trajectory
+ * @param max_allowable_acceleration [in] The maximum allowable acceleration along the
+ * trajectory. In m/s^2.
  *
- * @param max_allowable_acceleration [in] The max allowable acceleration at any discrete
- * point on the trajectory
+ * @param speed_cap [in] The maximum allowable speed at any point on the trajectory. This
+ * value is defined by the user and this speed limit will not be exceeded even if possible
+ * given the physical possibility. In m/s.
  *
+ * @param max_allowable_speed_profile [in] The array that will be modified in-place to
+ * contain the absolute maximum allowable speed profile. In m/s.
  */
-void app_trajectory_planner_getMaxAllowableSpeedProfile(
-    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    Polynomial2dOrder3_t path, const unsigned int num_elements,
-    ArcLengthParametrization_t arc_length_parameterization,
-    const float arc_segment_length, const float max_allowable_acceleration);
+void app_trajectory_planner_getAbsoluteMaximumSpeedProfile(
```

Comment:
What does absolute mean in this context?

Reply:
Originally I named it 'absolute' because it incorporated the user-set maximum speed profile - unlike the old one which was just curvature based. I'll rename it 👍